src/provider/poseidon/poseidon_inner.rs

```diff
@@ -0,0 +1,490 @@
+use std::marker::PhantomData;
+
+use ff::PrimeField;
+use generic_array::{sequence::GenericSequence, typenum, ArrayLength, GenericArray};
+use typenum::*;
+
+use super::{
+  matrix::transpose,
+  mds::{derive_mds_matrices, factor_to_sparse_matrixes, generate_mds},
+  preprocessing::compress_round_constants,
+};
+
+use super::{
+  hash_type::HashType,
+  matrix::{left_apply_matrix, Matrix},
+  mds::{MdsMatrices, SparseMatrix},
+  quintic_s_box, round_constants, round_numbers, Strength, DEFAULT_STRENGTH,
+};
+
+/// Available arities for the Poseidon hasher.
+pub trait Arity<T>: ArrayLength {
```

Comment:
Since Nova codebase only uses the Sponge type, do we need support for different Arity?

Reply:
I think we can remove the "`Arity` option", and just hard code whatever `Arity` the `Nova` codebase uses?

Reply:
Nova codebase does not explicitly set the arity and we only use the sponge APIs. Do you know how the sponge code selects arity?

Reply:
From what I can see, passing `U24` as the second type parameter to the `Sponge`'s `api_constants` method sets the "arity" of the sponge.

https://github.com/microsoft/Nova/blob/8136faa864874440fc44cbd373d4b4de813338bd/src/provider/poseidon.rs#L30